io/io/inc/TFile.h

```diff
@@ -281,9 +281,9 @@ class TFile : public TDirectoryFile {
    virtual void        Seek(Long64_t offset, ERelativeTo pos = kBeg);
    virtual void        SetCacheRead(TFileCacheRead *cache, TObject *tree = nullptr, ECacheAction action = kDisconnect);
    virtual void        SetCacheWrite(TFileCacheWrite *cache);
-   virtual void        SetCompressionAlgorithm(Int_t algorithm = ROOT::RCompressionSetting::EAlgorithm::kUseGlobal);
-   virtual void        SetCompressionLevel(Int_t level = ROOT::RCompressionSetting::ELevel::kUseMin);
-   virtual void        SetCompressionSettings(Int_t settings = ROOT::RCompressionSetting::EDefaults::kUseCompiledDefault);
+   virtual void        SetCompressionAlgorithm(Int_t algorithm = (Int_t)ROOT::RCompressionSetting::EAlgorithm::kUseGlobal);
```

Comment:
If we are deprecating the int values, we should consider deprecating the int signature (only for the algorithm)
```suggestion
   virtual void        SetCompressionAlgorithm(Int_t algorithm = (Int_t)ROOT::RCompressionSetting::EAlgorithm::kUseGlobal) R__DEPRECATED(6, 34, "Pass a ROOT::RCompressionSetting::EAlgorithm instead of an int");
   virtual void        SetCompressionAlgorithm(ROOT::RCompressionSetting::EAlgorithm algorithm = ROOT::RCompressionSetting::EAlgorithm::kUseGlobal);
```